swap/src/env.rs

```diff
@@ -25,6 +25,22 @@ impl Config {
     pub fn monero_sync_interval(&self) -> Duration {
         sync_interval(self.monero_avg_block_time)
     }
+
+    pub fn with_bitcoin_finality_confirmations(
+        mut self,
+        bitcoin_finality_confirmations: u32,
+    ) -> Config {
+        self.bitcoin_finality_confirmations = bitcoin_finality_confirmations;
+        self
+    }
```

Comment:
The fields are all public which means this function doesn't provide a whole lot of value given that there is no encapsulation or data hiding going on.

Consequently, the main decision on whether or not this is a good design comes down to a trade-off of ergonomics on the caller side vs the amount of code we need to add to make it happen.

Given all fields are public, we can also write (on the caller side):

```rust
let config = Config {
	bitcoin_finality_confirmations: 10,
	..config
};
```

without needing these functions.

Is this readable enough or do the "builder" functions add enough value for the amount code that is being added?

I am tempted to say we don't need the builder functions but I'll leave it up to you :)